cpp/src/arrow/testing/gtest_util.cc

```diff
@@ -1043,8 +1041,8 @@ std::shared_ptr<Array> ExampleDictExtension() {
 }
```

Comment:
The `DictArrayFromJSON` in `ExampleDictExtension` above could also be moved to `json::DictArrayFromJSONString`.